inprocess_cache.go

```diff
@@ -2,20 +2,22 @@ package cache
 
 import (
 	"errors"
+	"fmt"
 	"time"
 
 	"github.com/shaj13/libcache"
 	_ "github.com/shaj13/libcache/lru"
 )
 
+var _ Cache = (*memLibCache)(nil)
+
 type memLibCache struct {
 	// caveat:
 	// libcache won't evict upon timeout, eviction only happens when it's expired + the cache key is read
 	cache libcache.Cache
-	Cache
 }
 
-func NewLibcache(cap int, ttl time.Duration) (Cache, error) {
+func NewLibcache(cap int, ttl time.Duration) (*memLibCache, error) {
```

Comment:
returning a struct allows us to extend this particular implementation with extra functionality, for example we could add a new method just for this cache and be able to call it

Imagine wanting to reset the cache in a test, you could add a `Prune` method outside of the interface
```go
func(c *memLibCache) Prune() {  /* delete all */ }
```
And the client can still use it in tests
```go
c := NewLibcache()
c.Prune() // this is not possible returning the interface
```